MINI_2016/Homeworks/Second/PiotrKamoda.Rmd

```diff
@@ -0,0 +1,171 @@
+---
+title: "hw2"
+author: "Piotr Kamoda"
+date: "22 pazdziernika 2016"
+output: html_document
+---
+
+```{r setup, echo=FALSE}
+library(caret)
+d1=read.table("student-mat.csv",sep=";",header=TRUE)
+```
+
+We have been given following data:
+
+```{r data}
+head(d1)
+````
+
+The task is to choose variables for knn and find optimal k. Lets start with dividing data:
+
+```{r divide}
+set.seed(1313)
+indxTrain <- createDataPartition(y = d1$Dalc, p = 0.75)
+str(indxTrain)
+
+d1Train <- d1[indxTrain$Resample1,]
+d1Test <- d1[-indxTrain$Resample1,]
+d1Train$Dalc <- factor(d1Train$Dalc) #Make sure setup is ok
+d1Test$Dalc <- factor(d1Test$Dalc) #Make sure setup is ok
+```
+
+I've chosen Dalc as Factor variable and failures and absences as reference data. Now lets find optimal k for that:
+
```

Comment:
I have told it to students as a summary of your homework, we can talk about your problems at our labs. Best